config/src/config/mod.rs

```diff
@@ -336,11 +336,18 @@ impl NodeConfig {
 
         self.indexer.starting_version = match std::env::var("STARTING_VERSION").ok() {
             None => self.indexer.starting_version,
-            Some(s) => Some(s.parse().map_err(|_| {
-                Error::InvariantViolation(
-                    "Could not parse 'STARTING_VERSION' env var as u64".to_string(),
-                )
-            })?),
+            Some(s) => match s.parse::<u64>() {
+                Ok(version) => Some(version),
+                Err(_) => {
+                    // Doing this instead of failing. This will allow a processor to have STARTING_VERSION: undefined when deploying
+                    aptos_logger::warn!(
+                        "Invalid STARTING_VERSION: {}, using {:?} instead",
```

Comment:
Can we make a todo to alarm on this, so it doesn't go unnoticed for too long?

Reply:
Oh this will not be unnoticed lol. This will only show up during startup and it'll be clear to the person running this whether we're starting from the right version.